os/arch/arm/src/artik053/src/artik053_alc5658.c

```diff
@@ -0,0 +1,341 @@
+		g_alc5658info.handler = NULL;
+		g_alc5658info.arg = NULL;
+	}
+
+	return OK;
+}
+
+static bool alc5658_enable(FAR const struct alc5658_lower_s *lower, bool enable)
+{
+	static bool enabled;
+	irqstate_t flags;
+	bool ret;
+
+	/* Has the interrupt state changed */
+
+	flags = irqsave();
+	if (enable != enabled) {
+		/* Enable or disable interrupts */
+
+		if (enable && g_alc5658info.handler) {
+			audvdbg("Enabling\n");
+			/* DO IRQ ENABLING HERE */
+			enabled = true;
+		} else {
+			audvdbg("Disabling\n");
+			/* DO IRQ DISABLING HERE */
+			enabled = false;
+		}
+	}
+
+	ret = enabled;
+	irqrestore(flags);
+	return ret;
+}
+
+static int alc5658_interrupt(int irq, FAR void *context)
+{
+	/* Just forward the interrupt to the ALC5658 driver */
+
+	audvdbg("handler %p\n", g_alc5658info.handler);
+	if (g_alc5658info.handler)
+		return g_alc5658info.handler(&g_alc5658info.lower, g_alc5658info.arg);
+
+	/* We got an interrupt with no handler.  This should not
+	 * happen.
+	 */
+
+	return OK;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: s5j_alc5658_initialize
+ *
+ * Description:
+ *   This function is called by platform-specific, setup logic to configure
+ *   and register the ALC5658 device.  This function will register the driver
+ *   as /dev/alc5658[x] where x is determined by the minor device number.
+ *
+ * Input Parameters:
+ *   minor - The input device minor number
+ *
+ * Returned Value:
+ *   Zero is returned on success.  Otherwise, a negated errno value is
+ *   returned to indicate the nature of the failure.
+ *
+ ****************************************************************************/
+int s5j_alc5658_initialize(int minor)
+{
+	FAR struct audio_lowerhalf_s *alc5658;
+	FAR struct audio_lowerhalf_s *pcm;
+	FAR struct i2c_dev_s *i2c;
+	FAR struct i2s_dev_s *i2s;
+	static bool initialized;
+	char devname[12];
+	int ret;
+
+	audvdbg("minor %d\n", minor);
+	DEBUGASSERT(minor >= 0 && minor <= 25);
+
+	/* Have we already initialized?  Since we never uninitialize we must prevent
+	 * multiple initializations.  This is necessary, for example, when the
+	 * touchscreen example is used as a built-in application in NSH and can be
+	 * called numerous time.  It will attempt to initialize each time.
+	 */
+
+	if (!initialized) {
+		/* Configure the ALC5658 interrupt pin */
+
+		/* Do we have an interrupt PIN ? FIXME */
+
+		/* Get an instance of the I2C interface for the ALC5658 chip select */
+
+		i2c = up_i2cinitialize(CONFIG_ALC5658_I2C_PORT);
+		if (!i2c) {
+			auddbg("ERROR: Failed to initialize CODEC I2C%d\n");
+			ret = -ENODEV;
+			goto errout;
+		}
+
+		/* Get an instance of the I2S interface for the ALC5658 data channel */
+
+		i2s = s5j_i2s_initialize();
+		if (!i2s) {
```

Comment:
It should be called with 0 as argument, 0 indicating the the i2s port.

Reply:
Fixed.